test/dattri/algorithm/test_rps.py

```diff
@@ -0,0 +1,117 @@
+"""Test for RPS."""
+
+import torch
+from torch import nn
+from torch.nn import functional
+from torch.utils.data import DataLoader, TensorDataset
+
+from dattri.algorithm.rps import RPSAttributor
+from dattri.benchmark.utils import SubsetSampler
+
+
+class MLP(nn.Module):
+    """A multi-layer perceptron model for MNIST dataset."""
+
+    def __init__(
+        self,
+        input_size=28 * 28,
+        hidden_size1=128,
+        hidden_size2=64,
+        num_classes=10,
+    ):
+        """Initialize the multi-layer perceptron model."""
+        super(MLP, self).__init__()
+        self.fc1 = nn.Linear(input_size, hidden_size1)
+        self.fc2 = nn.Linear(hidden_size1, hidden_size2)
+        self.fc3 = nn.Linear(hidden_size2, num_classes)
+
+    def forward(self, x):
+        """Forward pass of the multi-layer perceptron model."""
+        x = x.view(x.size(0), -1)
+        x = torch.relu(self.fc1(x))
+        x = torch.relu(self.fc2(x))
+        return self.fc3(x)
+
+
+class TestRPS:
+    """Test for RPS."""
+
+    def test_rps_multi(self):
+        """Test for RPS."""
+        # multi-class
+        # init the model
+        model = MLP(num_classes=10)
+        dataset = TensorDataset(torch.randn(10, 1, 28, 28), torch.randint(0, 10, (10,)))
+        full_train_loader = DataLoader(dataset, batch_size=2)
+        test_loader = DataLoader(dataset, batch_size=2)
+        train_loader = DataLoader(
+            dataset,
+            batch_size=2,
+            sampler=SubsetSampler(range(4)),
+        )
+
+        # define the loss function
+        def f(pre_activation_list, label_list):
+            return functional.cross_entropy(pre_activation_list, label_list)
+
+        # define the RPS attributor w/ cache
+        attributor = RPSAttributor(
+            loss_func=f,
+            model=model,
+            final_linear_layer_name="fc3",
```

Comment:
this layer name is kind of different from `list(model.named_paramters()).keys()` right? I guess the keys might be `fc3.weights` and `fc3.bias` for this layer. We may need a better explanation in the docuemnt.

Reply:
Yeah, now it is the "layer name" but not "parameter name". Now for the fine-tuning step, I only choose to do it on the weight matrix (ignoring the bias entirely). If this is really a matter, we can do the "add-1-dim" trick to incorporate the bias into the weight matrix first then do the fine-tuning.